buildSrc/subprojects/parser/src/main/java/org/apache/calcite/buildtools/parser/ExtractedData.java

```diff
@@ -33,9 +33,9 @@
   public final List<String> tokenAssignments;
 
   public ExtractedData() {
-    keywords = new LinkedHashMap<Keyword, String>();
-    nonReservedKeywords = new HashSet<Keyword>();
-    functions = new LinkedHashMap<String, String>();
-    tokenAssignments = new LinkedList<String>();
+    keywords = new LinkedHashMap<>();
+    nonReservedKeywords = new HashSet<>();
+    functions = new LinkedHashMap<>();
+    tokenAssignments = new ArrayList<>();
```

Comment:
Just wondering, why the change to ArrayList throughout the code?

Reply:
Our linter says it's more efficient than LinkedList.